internal/config/from_env.go

```diff
@@ -34,19 +35,29 @@ type envConfig struct {
 }
 
 func (c *envConfig) Hosts() ([]string, error) {
-	hasDefault := false
 	hosts, err := c.Config.Hosts()
-	for _, h := range hosts {
-		if h == ghinstance.Default() {
-			hasDefault = true
+	if err != nil {
+		return nil, err
+	}
+
+	hostSet := set.NewStringSet()
+	hostSet.AddValues(hosts)
+
+	// If GH_HOST is set and there is a valid environment variable
+	// token for the host then add it to list.
+	if host := os.Getenv(GH_HOST); host != "" {
+		if token, _ := AuthTokenFromEnv(host); token != "" {
```

Comment:
If GH_HOST is set in the environment, would it be a good idea to make it the default host even if GH_ENTERPRISE_TOKEN wasn't set?

Reply:
Can you elaborate on what you mean by default host? Are you suggesting that if `GH_HOST` is set then it should always be returned or are you suggesting that if `GH_HOST` is set then it should be the only value returned?

Reply:
So, what I meant was, I see two conditions here for `hostSet.Add(<GH_HOST>)` to happen: that GH_HOST is set and that GH_ENTERPRISE_TOKEN is also set. Could we remove the second condition and just do:
```
if GH_HOST is set
  hostSet.Add(<GH_HOST>)
```

Reply:
Yeah that makes sense to me. Updated.